src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/BookBinder.java

```diff
@@ -0,0 +1,132 @@
+                EnchantmentStorageMeta itemMeta = (EnchantmentStorageMeta) item.getItemMeta();
+                EnchantmentStorageMeta targetMeta = (EnchantmentStorageMeta) target.getItemMeta();
+
+                Map<Enchantment, Integer> storedItemEnchantments = itemMeta.getStoredEnchants();
+                Map<Enchantment, Integer> storedTargetEnchantments = targetMeta.getStoredEnchants();
+                enchantments.putAll(storedItemEnchantments);
+
+                for (Map.Entry<Enchantment, Integer> entry : storedTargetEnchantments.entrySet()) {
+                    enchantments.merge(entry.getKey(), entry.getValue(), (a, b) -> {
+                            if (a == b) {
+                                if (hasCustomMaxLevel.getValue()) {
+                                    return a + 1 > customMaxLevel.getValue() ? customMaxLevel.getValue() : a + 1; 
+                                } else {
+                                    return a + 1;
+                                }
+                                
+                            } else {
+                                if (hasCustomMaxLevel.getValue()) {
+                                    return Math.max(a, b) > customMaxLevel.getValue() ? customMaxLevel.getValue() : Math.max(a, b);
+                                } else {
+                                    return Math.max(a, b);
+                                }
+                               
+                            }
+                    });
+                }
+
+                if (enchantments.size() > 0) {
+                    ItemStack book = new ItemStack(Material.ENCHANTED_BOOK);
+                    book.setAmount(1);
+
+                    EnchantmentStorageMeta enchantMeta = (EnchantmentStorageMeta) book.getItemMeta();
+                    for (Map.Entry<Enchantment, Integer> entry : enchantments.entrySet()) {
+                        if (bypassVanillaMaxLevel.getValue()) {
+                            enchantMeta.addStoredEnchant(entry.getKey(), entry.getValue(), true);
+                        } else {
+                            enchantMeta.addStoredEnchant(entry.getKey(), entry.getValue(), false);
+                        }
+                    }
+
+                    book.setItemMeta(enchantMeta);
+                    
+                    MachineRecipe recipe = new MachineRecipe(25 * enchantments.size() / this.getSpeed(), new ItemStack[] {target, item}, new ItemStack[] {book});
+
+                    if (!InvUtils.fitAll(menu.toInventory(), recipe.getOutput(), getOutputSlots())) {
+                        return null;
+                    }
+
+                    for (int inputSlot : getInputSlots()) {
+                        menu.consumeItem(inputSlot);
+                    }
+
+                    return recipe;
+                }
+
+                return null;
+            }
+
+        }
+
+        return null;
+    }
+
+    private boolean isCompatible(ItemStack item) {
```

Comment:
You should add an annotation here
```suggestion
    private boolean isCompatible(@Nullable ItemStack item) {
```